### ISSUE TITLE
Bug: `prefer-query-by-disappearance` reports error in an incorrect part of the code

### ISSUE THREAD

**Issue by Belco90** [testing-library/eslint-plugin-testing-library#600]
### Have you read the Troubleshooting section?

Yes

### Plugin version

v5.5.0

### ESLint version

v8.14.0

### Node.js version

v16.15.0

### package manager and version

npm v8.9.0

### Operating system

macOS v12.3.1

### Bug description

When there is an error reported by `prefer-query-by-disappearance`, it's displayed at the very beginning of the callback passed to `waitForElementToBeRemoved`. Instead, the error should be placed in the query within the callback.

### Steps to reproduce

```js
it('should do something', async () => {
  render(<App />)

  fireEvent.click(screen.getByRole('button'))

  await waitForElementToBeRemoved(() => screen.getByText('Loading'))
})
```

### Error output/screenshots

![CleanShot 2022-06-03 at 01 23 52](https://user-images.githubusercontent.com/2677072/171753913-770774a8-708e-452a-a338-d823f5598c3c.png)

### ESLint configuration

N/A

### Rule(s) affected

`prefer-query-by-disappearance`

### Do you want to submit a pull request to fix this bug?

No

**Comment by sjarva** [comment]
Thanks @Belco90  for tagging this as a "good first issue", it encouraged me to contribute to this repo! 🤗

**Comment by Belco90** [comment]
@sjarva Great! Thanks for your contribution, I'll try to review it soon.
